### PR TITLE
Query: Perf: Split entity shapers into multiple optimized implementat…

### PR DESCRIPTION
**Pull request by anpete**
…ions.

- Moved shapers to .Internal

### CODE REVIEW

**Comment by anpete** [comment]
@maumar

File: src/EntityFramework.Relational/Query/ExpressionVisitors/Internal/EntityShaper.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            [NotNull] KeyValueFactory keyValueFactory,
+            [NotNull] Func<ValueBuffer, object> materializer)
+            : base(querySource)
+        {
+            EntityType = entityType;
+            KeyValueFactory = keyValueFactory;
+            Materializer = materializer;
+        }
+
+        protected virtual string EntityType { get; }
```
Comment (maumar) [inline src/EntityFramework.Relational/Query/ExpressionVisitors/Internal/EntityShaper.cs]: EntityTypeName?

Comment (anpete) [inline src/EntityFramework.Relational/Query/ExpressionVisitors/Internal/EntityShaper.cs]: Will do.

**Comment by maumar** [comment]
:shipit:

**Pull request closed without merging**